clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -744,34 +798,17 @@ public void testFencedMemberStopHeartbeatUntilItReleasesAssignmentToRejoin() {
         verify(heartbeatRequestState).onFailedAttempt(anyLong());
         verify(heartbeatRequestState).reset();
 
+        when(heartbeatRequestState.canSendRequest(anyLong())).thenReturn(false);
         when(membershipManager.state()).thenReturn(MemberState.FENCED);
         result = heartbeatRequestManager.poll(time.milliseconds());
         assertEquals(0, result.unsentRequests.size(), "Member should not send heartbeats while FENCED");
 
+        when(heartbeatRequestState.canSendRequest(anyLong())).thenReturn(true);
         when(membershipManager.state()).thenReturn(MemberState.JOINING);
         result = heartbeatRequestManager.poll(time.milliseconds());
         assertEquals(1, result.unsentRequests.size(), "Fenced member should resume heartbeat after transitioning to JOINING");
     }
 
-    @ParameterizedTest
-    @ApiKeyVersionsSource(apiKey = ApiKeys.CONSUMER_GROUP_HEARTBEAT)
-    public void testSendingLeaveGroupHeartbeatWhenPreviousOneInFlight(final short version) {
```

Comment:
uhm we shouldn't remove this test (it was actually an issue we discovered, fixed, and added the test for coverage). I see it involves checking the HB content (that is might be off just because the heartbeatState is a mock, but give it another try after addressing the comments above to turn that state into an actual instance able to build the HB content)